README.md

```diff
@@ -86,6 +86,169 @@ server.enableAuth('user', 'pass');
+serverHandler = server.listen(8088, 'localhost');
+
+client = rpc.Client.$create(8088, 'localhost');
+client.basicAuth('user', 'pass');
+
+var params = ['Hello!, Authorization!'];
+client.call('echo', params, function (err, result) {
+    if (err) { /* do something with error */ }
+    // do something with result
+    console.log(result); // in this case, 'Hello, Authorization!'
+});
+
+serverHandler.close();
+server = null;
+client = null;
+```
+
+Cookie Authorization Communication
+
+```js
+var rpc = require('json-rpc2');
+
+var server = rpc.Server.$create({
+    'websocket': true
+});
+
+server.define('echo', function (opts, args, callback) {
+    // function callback(error, result)
+    callback(null, args[0]);
+});
+
+server.enableCookieAuth(function (cookieValue) {
+    // do call to user service or pass it to middleware
+    return (cookieValue === 'validCookieValue');
+});
+serverHandler = server.listen(8088, 'localhost');
+
+client = rpc.Client.$create(8088, 'localhost');
+client.cookieAuth('validCookieValue');
+
+var params = ['Hello!, Authorization!'];
+client.call('echo', params, function (err, result) {
+    if (err) { /* do something with error */ }
+    // do something with result
+    console.log(result); // in this case, 'Hello, Authorization!'
+});
+
+serverHandler.close();
+server = null;
+client = null;
+```
+
+Cookie Authorization Communication
+
+```js
+var rpc = require('json-rpc2');
+
+var server = rpc.Server.$create({
+    'websocket': true
+});
+
+server.define('echo', function (opts, args, callback) {
+    // function callback(error, result)
+    callback(null, args[0]);
+});
+
+server.enableJWTAuth(function (tokenValue) {
+    // do call to user service or pass it to middleware
+    return (cookieValue === 'validCookieValue');
+});
+serverHandler = server.listen(8088, 'localhost');
+
+client = rpc.Client.$create(8088, 'localhost');
+client.jwtAuth('validTokenValue');
+
+var params = ['Hello!, Authorization!'];
+client.call('echo', params, function (err, result) {
+    if (err) { /* do something with error */ }
+
+    // do something with result
+    console.log(result); // in this case, 'Hello, Authorization!'
+});
+
+serverHandler.close();
+server = null;
+client = null;
+```
+
+Switching between Authorization Methods
+
+```js
+var rpc = require('json-rpc2');
+
+var server = rpc.Server.$create({
+    'websocket': true
+});
+
+server.define('echo', function (opts, args, callback) {
+    // function callback(error, result)
+    callback(null, args[0]);
+});
+
+server.listen(8088, 'localhost');
+
+// Server Authorization
+server.enableBasicAuth('user', 'pass');
+server.enableCookieAuth(function (cookieValue) {
+    // handle cookie validity through a user service or middleware
+    return (cookieValue === 'validCookieValue');
+});
+server.enableJWTAuth(function (tokenValue) {
+    // handle JWT token validity through a user service or middleware
+    return (tokenValue === 'validTokenValue');

```

Comment:
Q has the same API for try (https://github.com/kriskowal/q/wiki/API-Reference). In both cases, the functions should throw an error in case of non-success, return the value in case of success.

I rewrote the promisify way (called promise.nodeify) for a function with a callback in a short gist 
https://gist.github.com/danielmihai/697074f3414e8e71c70dfbd1fbcc7617 .

Same functionality can be implemented using Bluebird Promise. Choose one and I'll make it stick. I'm more fluent in using Q, but I also used Bluebird and Promise a long time ago (with Angular, I think).